### DISCUSSION TITLE
Context getting Deleted Site

### DISCUSSION THREAD

**Discussion by anthonywhite** [pnp/pnpcore#659] [Q&A] [answered]
We have a Durable Azure function that we use to create sites and apply templates. We based this on the [PnP Azure Function v3 sample](https://pnp.github.io/pnpcore/demos/Demo.AzureFunction/README.html) i.e. using dependency injection with `PnP.Core.Services.Builder` to set up the authentication provider and `PnPCoreSiteOptions` etc.

In recent testing we came across an issue where a PnPContext created with a Site Url is pointing to a deleted site, even when the site has been re-created after emptying the tenant recycle bin.

The context is always created with a using statement like so:
```
using (var newPnPContext = await pnpContextFactory.CreateAsync(new Uri(<SiteUrl>)))
{
// do stuff
}
```
So it feels like the underlying connection somehow needs to be re-made because it has the old site GUID, like in PowerShell we might execute a `New-PnPConnection` command. Is this correct and is there some way of doing this in our function?

**Comment by jansenbe** [comment, marked as answer]
@anthonywhite : when we create a context using one of the CreateAsync method there are always two REST calls made to the site collection in question...so if you were able to create the `PnPContext` for a site collection that means the site collection was not deleted. If the site collection would have been delete or resided in the site collection recycle bin this operation would have failed. 

If you created a context before the site collection was deleted then the context use that triggers a call to SharePoint Online would have resulted in an exception thrown.

**Comment by anthonywhite** [comment]
Hi @jansenbe , thanks for your response. I was expecting the behaviour to be what you said, but the mystery is the site was deleted well before the function run instance (so before context creation). Unless the function runtime is holding an old object?

**Reply by jansenbe** [reply]
@anthonywhite : The `PnPContext` objects are not added the dependency container and they're not reused. Seems like the durable function somehow "dehydrated" it's state and continued execution while in the meanwhile the site collection got deleted? Anyhow, using the PnPContext at that point would result in an exception being thrown.

**Reply by jansenbe** [reply]
Regardless of that, it always can happen that a site collection gets deleted while some other process has a `PnPContext` in use for it.

**Reply by jansenbe** [reply]
@anthonywhite : are you good with the provided answer? If so then feel free to make this as answered.

**Reply by anthonywhite** [reply]
Hi @jansenbe we are still seeing this issue but I have not had time to fully repro and debug. It may be a functions behaviour, but I can't say yet.
